pkg/plugins/manager/process/process.go

```diff
@@ -123,7 +67,7 @@ func restartKilledProcess(ctx context.Context, p *plugins.Plugin) error {
 			if err := ctx.Err(); err != nil && !errors.Is(err, context.Canceled) {
 				return err
 			}
-			return nil
+			return p.Stop(ctx)
```

Comment:
Could there be an error and the plugin continues to run/are not stopped? Or will `<Store service>.shutdown(ctx)` be called after this returns when plugin already stopped?

Reply:
>Could there be an error and the plugin continues to run/are not stopped?

This will only happen when `errors.Is(err, context.Canceled)`, which I _think_ makes sense to try stop the plugin here. If it fails here, then `Store.shutdown` will be called on Grafana shutdown. 

>will <Store service>.shutdown(ctx) be called after this returns when plugin already stopped?

It could, but calling Stop() multiple times is idempotent.